src/transformers/models/fastvit/configuration_fastvit.py

```diff
@@ -0,0 +1,152 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" FastViT model configuration"""
+
+from collections import OrderedDict
+from typing import Mapping
+
+from packaging import version
+
+from ...configuration_utils import PretrainedConfig
+from ...onnx import OnnxConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+FASTVIT_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "apple/fastvit-t8": "https://huggingface.co/apple/fastvit-t8/resolve/main/config.json",
+    "apple/fastvit-t12": "https://huggingface.co/apple/fastvit-t12/resolve/main/config.json",
+    "apple/fastvit-s12": "https://huggingface.co/apple/fastvit-s12/resolve/main/config.json",
+    "apple/fastvit-sa12": "https://huggingface.co/apple/fastvit-sa12/resolve/main/config.json",
+    "apple/fastvit-sa24": "https://huggingface.co/apple/fastvit-sa24/resolve/main/config.json",
+    "apple/fastvit-sa36": "https://huggingface.co/apple/fastvit-sa36/resolve/main/config.json",
+    "apple/fastvit-ma36": "https://huggingface.co/apple/fastvit-ma36/resolve/main/config.json",
+    # See all ViT models at https://huggingface.co/models?filter=fastvit
+}
+
+
+class FastViTConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`FastViTModel`]. It is used to instantiate an
+    FastViT model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the FastViT
+    [google/fastvit-base-patch16-224](https://huggingface.co/google/fastvit-base-patch16-224) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        image_size (`int`, *optional*, defaults to `256`):
+            The size (resolution) of each image.
+        num_channels (`int`, *optional*, defaults to `3`):
+            The number of input channels.
+        patch_size (`int`, *optional*, defaults to `4`):
+            The size (resolution) of each patch.
+        depths (`List[int]`, *optional*, defaults to `[2, 2, 4, 2]`):
+            The number of Token Mixer blocks in each FastViTLayer Block.
+        num_attention_heads (`int`, *optional*, defaults to 32):
+            Number of attention heads for each attention layer in the Transformer encoder.
+        hidden_sizes (`List[int]`, *optional*, defaults to `[48, 96, 192, 384]`):
+            Dimensionality of the encoder layers in each FastViTLayer Block.
+        pos_embeds (`List[Bool]`, *optional*, defaults to `False`, equivalent to `[False, False, False, False]`)
```

Comment:
Need to add a comment as a reminder, you can write None or RepCPE for Conditional Position Encoding